File: src/ManagementPlatform.Domain/OutboxMessage.cs
namespace ManagementPlatform.Domain;

public sealed class OutboxMessage
{
    public long Id { get; set; }
    public long CheckoutAttemptId { get; set; }
    public CheckoutAttempt CheckoutAttempt { get; set; } = null!;
    public OutboxMessageType Type { get; set; }
    public OutboxStatus Status { get; set; } = OutboxStatus.Pending;
    public string PayloadJson { get; set; } = "{}";
    public int AttemptCount { get; set; }
    public string? LastError { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset? NextAttemptAt { get; set; }
    public DateTimeOffset? LockedAt { get; set; }
    public DateTimeOffset? ProcessedAt { get; set; }
}

File: src/main/java/com/managementplatform/domain/model/OutboxMessage.java
package com.managementplatform.domain.model;

import com.managementplatform.domain.enums.OutboxMessageType;
import com.managementplatform.domain.enums.OutboxStatus;
import java.time.Instant;

public record OutboxMessage(
    long id,
    long checkoutAttemptId,
    OutboxMessageType type,
    OutboxStatus status,
    String payloadJson,
    int attemptCount,
    String lastError,
    Instant createdAt,
    Instant nextAttemptAt,
    Instant lockedAt,
    Instant processedAt,
    DeadLetterMessage deadLetterMessage
) {
}
